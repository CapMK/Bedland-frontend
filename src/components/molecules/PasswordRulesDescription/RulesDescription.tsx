import * as Styled from './RulesDescription.styled';

const RulesDescription = () => (
  <Styled.RulesDescriptionContainer>
    <span> Remember, your password must include:</span>{' '}
    <Styled.OrderedRulesDesc style={{ listStyleType: 'disc' }}>
      <li>8-24 characters</li>
      <li>At least one uppercase character</li>
      <li>At least one special character</li>
    </Styled.OrderedRulesDesc>
  </Styled.RulesDescriptionContainer>
);
export default RulesDescription;
