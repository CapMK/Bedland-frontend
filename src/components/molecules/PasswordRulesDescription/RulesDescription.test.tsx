import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import RulesDescription from './RulesDescription';

describe('RulesDescription', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const snapshot = renderer
      .create(
        <BrowserRouter>
          <RulesDescription />
        </BrowserRouter>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
