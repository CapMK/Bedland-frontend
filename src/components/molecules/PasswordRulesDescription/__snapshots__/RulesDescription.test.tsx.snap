// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RulesDescription > renders correctly 1`] = `
.c0 {
  padding: 0rem 2px 0rem 2.9375rem;
  width: 100%;
}

.c1 {
  liststyletype: disc;
  margin-left: 10px;
  padding: 0 0 0 1rem;
}

<div
  className="c0"
>
  <span>
     Remember, your password must include:
  </span>
   
  <ol
    className="c1"
    style={
      {
        "listStyleType": "disc",
      }
    }
  >
    <li>
      8-24 characters
    </li>
    <li>
      At least one uppercase character
    </li>
    <li>
      At least one special character
    </li>
  </ol>
</div>
`;
